### DISCUSSION TITLE
JS not reload

### DISCUSSION THREAD

**Discussion by Pratikginoya** [livewire/livewire#8565] [1. Help]
I have added one common script in my "footer.blade.php"
For ex:- $('.my-class').css('color','red');

Now when I do any CRUD from my component then this CRUD data has been reload/change only but not reload the scripts which i have used in "footer.blade.php" for this component blade file....

For Ex:-
my-component.blade.php

<div>
    <div class="my-class"> HELLO</div>
</div>

Note :- Currently font-color of HELLO is "Red" as per my footer script on page load

now i edit this data from HELLO to GOOD_MORNING in "MyComponent" hence my new data is,

<div>
    <div class="my-class"> GOOD_MORNING </div>
</div>

but now font-color is "Black" instead of "Red" because my footer.blade.php script is not called on render....
if i reload the page then the font-color is "Red" because on page-load or navigate my footer script has been called....

How to fix it..? (I want to reload javascript/jQuery which i have used in footer.blade.php on component re-render)
Please help to resolve this....

**Comment by H-Hp** [comment]
To ensure that the scripts in your `footer.blade.php` are executed whenever a component is re-rendered, you can use the `@once` directive in Laravel Blade along with the `@push('scripts')` and `@endpush` directives.

Here's how you can do it:

1. In your `app.blade.php` layout file (or the file that includes the `footer.blade.php`), add the following code where you want to include the scripts:

```html
@stack('scripts')
```

2. In your `footer.blade.php` file, wrap your scripts with `@once` and `@push('scripts')` directives like this:

```blade
@once
@push('scripts')
<script>
$(document).ready(function() {
    $('.my-class').css('color', 'red');
});
</script>
@endpush
@endonce
```

The `@once` directive ensures that the scripts are only pushed once to the stack, even if the component is re-rendered multiple times.

3. In your Livewire component's class, add the following method to make sure the scripts are re-rendered when the component updates:

```php
use Illuminate\Support\Facades\Blade;

public function render()
{
    Blade::renderOnce('path.to.your.footer', []);
    return view('livewire.my-component');
}
```

Replace `'path.to.your.footer'` with the actual path to your `footer.blade.php` file relative to the `resources/views` directory.

By doing this, whenever your Livewire component is re-rendered, the `footer.blade.php` file will be rendered as well, ensuring that the scripts within it are executed.

Alternatively, you can use the `@livewireScripts` directive provided by Livewire to include your scripts. Place the following code at the end of your `app.blade.php` layout file, just before the closing `</body>` tag:

```html
@livewireScripts
<script>
$(document).ready(function() {
    $('.my-class').css('color', 'red');
});
</script>
```

This method ensures that your scripts are included and executed whenever a Livewire component is re-rendered, without the need for the `@once` and `@push` directives.

**Reply by Pratikginoya** [reply]
Dear Sir/Ma'am.

Thanks for reply,
But still its not working, please refer below code for your reference,

this is my `app.blade.php`

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>@yield('title')</title>
  <link rel="stylesheet" href="{{asset('admin/plugins/bootstrap-colorpicker/css/bootstrap-colorpicker.min.css')}}">
  
  @livewireStyles
</head>
<body class="hold-transition sidebar-mini layout-fixed">
    @include('layouts/header')
  
     {{ $slot }}
    @yield('content') 

    @livewireScripts
    @include('layouts/footer')
    @include('layouts/scripts')
    @yield('scripts')
    <script>
      $(document).ready(function() {
        $('table td').css('color','red');
      });
    </script>

</body>
</html>

```

this is my `MyComponent` 

```
class MyComponent extends Component
{
    public $technologyData;
    protected $listeners = ['technologyAdded' => 'refreshTechnologyData'];

    public function render()
    {
        $this->technologyData = Technology::all();
        return view('livewire.technology-view',['technologyData',$this->technologyData]);
    }

    public function delete($id)
    {
        $techDelete = Technology::findOrFail($id);

        if($techDelete) {
            $techDelete->delete();
            session()->flash('success','Technology has been deleted succesfully.');
        }
    }

    public function refreshTechnologyData()
    {
        $this->technologyData = Technology::all();
        $this->dispatch('reloadAllScripts');
    }

    public function edit($id)
    {
        $this->dispatch('getEditDetailsOnCreate',$id);
    }
}
```

this is `my-component.blade.php` file

```
<div class="card card-primary">
  <div class="card-header">
    <h3 class="card-title">View Technology</h3>
  </div>

  @if (session()->has('success'))
    <div class="alert mt-3 mb-0" style="color: green;">
        {{ session('success') }}
    </div>
  @endif

  <div class="card-body">
    <table id="example2" class="table table-bordered table-hover">
      <thead>
      <tr>
        <th>Sr No</th>
        <th>Technology Name</th>
        <th>Status</th>
        <th>Edit Data</th>
        <th>Delete Data</th>
      </tr>
      </thead>
      
      @foreach($technologyData as $tech)
      <tr>
        <td>{{$loop->iteration}}</td>
        <td>{{$tech->tech_name}}</td>
        <td>
          <div class="custom-control custom-switch">
              <input type="checkbox" class="custom-control-input" id="statusChange{{$tech->id}}" {{$tech->status == 1 ? 'checked' : ''}}>
              <label class="custom-control-label" for="statusChange{{$tech->id}}"></label>
          </div>
        </td>
        <td><button wire:click="edit({{$tech->id}})" class="btn btn-primary">Edit</button></td>
        <td><button wire:confirm="Are you sure to delete?"  wire:click="delete({{$tech->id}})" class="btn btn-primary">Delete</button></td>
      </tr>
      @endforeach

    </table>
  </div>
</div>
```

I have doing the same as you mentioned in you reply but still its not working on component re-render....

Please help me to solve this....
Note :- I am using  `"livewire/livewire": "^3.5"`

Thank You.....

**Reply by juniorsanca** [reply]
Hello my friend @Pratikginoya did you solve this problem ? if yes, please how did you do ? thanks :)

**Reply by lrljoe** [reply]
> Hello my friend @Pratikginoya did you solve this problem ? if yes, please how did you do ? thanks :)

Please explain your use case.

**Reply by juniorsanca** [reply]
when i dispatch to a component my javascript do not work anymore on this component
